lua/entities/gmod_wire_expression2/base/compiler.lua

```diff
@@ -651,42 +651,57 @@ end
 
 function Compiler:InstrNUM(args)
 	self.prfcounter = self.prfcounter + 0.5
-	RunString("E2Lib.Compiler.native = function() return " .. args[3] .. " end")
-	return { Compiler.native }, "n"
+	local num = tonumber(args[3])
+
+	return {function()
+		return num
+	end}, "n"
 end
 
 function Compiler:InstrNUMI(args)
 	self.prfcounter = self.prfcounter + 1
-	Compiler.native = { 0, tonumber(args[3]) }
-	RunString("local value = E2Lib.Compiler.native E2Lib.Compiler.native = function() return value end")
-	return { Compiler.native }, "c"
+	local value = {0, tonumber(args[3])}
+
+	return {function()
+		return value
+	end}, "c"
 end
 
 function Compiler:InstrNUMJ(args)
 	self.prfcounter = self.prfcounter + 1
-	Compiler.native = { 0, 0, tonumber(args[3]), 0 }
-	RunString("local value = E2Lib.Compiler.native E2Lib.Compiler.native = function() return value end")
-	return { Compiler.native }, "q"
+	local value = {0, 0, tonumber(args[3]), 0}
+
+	return {function()
+		return value
+	end}, "q"
 end
 
 function Compiler:InstrNUMK(args)
 	self.prfcounter = self.prfcounter + 1
-	Compiler.native = { 0, 0, 0, tonumber(args[3]) }
-	RunString("local value = E2Lib.Compiler.native E2Lib.Compiler.native = function() return value end")
-	return { Compiler.native }, "q"
+	local value = {0, 0, 0, tonumber(args[3])}
+
+	return {function()
+		return value
+	end}, "q"
 end
 
 function Compiler:InstrSTR(args)
 	self.prfcounter = self.prfcounter + 1.0
-	RunString(string.format("E2Lib.Compiler.native = function() return %q end", args[3]))
-	return { Compiler.native }, "s"
+	local str = args[3]
```

Comment:
This looks okay: hopefully `%q` is the inverse of `dostring`, but if not then the old code is broken more than this PR's new code, because a string instruction's tokendata is supposed to be the [already-parsed string contents](https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/base/tokenizer.lua#L207).